logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -0,0 +1,66 @@
+# encoding: utf-8
+require "concurrent/atomic/atomic_fixnum"
+
+# This class goes hand in hand with the pipeline to provide a pool of
+# free workers to be used by pipeline worker threads. The pool is
+# internally represented with a SizedQueue set the the size of the number
+# of 'workers' the output plugin is configured with.
+#
+# This plugin also records some basic statistics
+module LogStash; class OutputDelegator
+  attr_reader :workers, :config, :worker_count
+
+  def initialize(logger, klass, *args)
```

Comment:
for readability and maintainability and especially when non-obvious, but ideally always, we should document method parameters. In particular here `*args` requires reading code below to understand the purpose.